tests/validators/date.test.ts

```diff
@@ -70,6 +70,10 @@ describe('DateValidator', () => {
 		test.each([new Date('2021-01-02'), new Date('2023-01-02')])('GIVEN %s THEN throws ConstraintError', (value) => {
 			expect(() => eqPredicate.parse(value)).toThrow(new ConstraintError('s.date.eq', 'Invalid Date value', value, 'expected = 2022-01-02'));
 		});
+
+		test.each(['abc', '', null, undefined])('GIVEN %s THEN throws ConstraintError', (value) => {
+			expect(() => eqPredicate.parse(value)).toThrow(new ConstraintError('s.date.ge', 'Expected a Date', value, 'expected = 2022-01-02'));
```

Comment:
This feels wrong, it shouldn't throw a ConstraintError, but a ValidationError.

Jest, why does your `.toThrow` suck so much?

Reply:
Also, the `test.each` should be moved to line 11, which is where it tests for invalid input.

Reply:
It does, but `.toThrow` literally only tests for `error.message`, doesn't check anything else.

The fact this passes **WITHOUT** instanceof checks is beyond my understanding,

Reply:
That's what they call design decisions. Nothing wrong there. Simply add a `.toBeInstanceOf` assertion before `.toThrow` and it's done. 

https://jestjs.io/docs/expect#tobeinstanceofclass

Reply:
Doesn't remove the fact it doesn't check the other properties. I'm leaning towards making our own macros calling `expect`.

Reply:
Regardless, this is a GitHub review comment. Not a place for you to vent and rant. Keep the review comments objective and useful please.